apps/namadillo/src/utils/index.ts

```diff
@@ -91,7 +91,9 @@ export const toDisplayAmount = (
   if (!displayUnit) {
     return baseAmount;
   }
-  return baseAmount.shiftedBy(-displayUnit.exponent);
+  return baseAmount
+    .shiftedBy(-displayUnit.exponent)
+    .decimalPlaces(6, BigNumber.ROUND_DOWN);
```

Comment:
we are using this function to set some value for any transfer, I'm a bit afraid of this imprecision spread to border cases. Shouldn't we apply this only on the visual components on the render phase?

Reply:
Changed. The error on Cosmos is because we can't estimate the gas required using an invalid channel. I've tried to fix that, but we don't have a right way to calculate a fallback for that. Before, the amount of gas used was hardcoded :/